ui/packages/catalog-ui-search/src/main/webapp/component/dropdown/query-src/dropdown.query-src.view.js

```diff
@@ -17,13 +17,73 @@ const Marionette = require('marionette')
 const _ = require('underscore')
 const $ = require('jquery')
 const DropdownView = require('../dropdown.view')
-const template = require('./dropdown.query-src.hbs')
 const ComponentView = require('../../query-src/query-src.view.js')
 const sources = require('../../singletons/sources-instance.js')
 const properties = require('../../../js/properties.js')
+import * as React from 'react'
+import { IntlProvider, FormattedMessage } from 'react-intl'
+
+const renderSourceLocal = source => (
+  <React.Fragment>
+    {source.local ? (
+      <i class="fa source-icon fa-home" />
+    ) : (
+      <i class="fa source-icon fa-cloud" />
+    )}
+  </React.Fragment>
+)
+
+const renderSourceAvailable = source => (
+  <React.Fragment>
+    <span className={'text-src ' + (source.available ? 'is-available' : '')}>
+      {source.available ? (
+        <i class="fa fa-exclamation-triangle src-availability" />
+      ) : null}
+      renderSourceLocal(source)
+      <span class="src-title">{source.id}</span>
+    </span>
+  </React.Fragment>
+)
 
 module.exports = DropdownView.extend({
-  template: template,
+  template(props) {
+    return (
+      <div>
+        <div className="dropdown-label">
+          <IntlProvider locale="en">
+            <FormattedMessage
+              id="sourcesKeyword"
+              defaultMessage={`{sources}`}
+              values={{ sources: properties.keywords['Sources'] }}
+            />
+          </IntlProvider>
+        </div>
+        <div className="dropdown-container">
+          <div className="dropdown-text is-input">
+            {props.enterprise ? (
+              <span className="text-src is-available">
+                <IntlProvider locale="en">
```

Comment:
Similar question as the Java side as far as requesting the data? Should we try to default to the user's locale and only if nothing is returned use the "en" replacements?